server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -223,7 +223,7 @@ public Query toQuery(Function input, LuceneQueryBuilder.Context context) {
             for (Reference nullableRef : ctx.nullableReferences()) {
                 // we can optimize with a field exist query and filter out all null values which will reduce the
                 // result set of the query
-                var refExistsQuery = IsNullPredicate.refExistsQuery(nullableRef, context, false);
+                var refExistsQuery = IsNullPredicate.refExistsQuery(nullableRef, context, true);
```

Comment:
> `array_length` spec

This should not affect the benchmarks since the change is strictly limited to `not` operator (`IsNullPredicate.refExistsQuery` is untouched).

Reply:
However, old behaviour was fast because it was wrong( skipped extra check with slow `GenericFunctionQuery`)

Reply:
Is there any chance we can improve the performance here again ?

Reply:
Actually, `!=[]` should be same as `array_length > 0`. 

With this PR we would diverge `array_length` and `!=[]` behaviour. 

https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/expression/scalar/ArrayUpperFunction.java#L219

Reply:
Pleas add a test for NULL element.
https://www.db-fiddle.com/f/qGpH7Aikj6icqY9n7R381E/2

`array_length` requires the same patch.

Reply:
> Actually, !=[] should be same as array_length > 0.

This isn't correct, has `array_length` is documented to return  `NULL` for empty arrays.
While `!=[]` should compare on an empty array instead of `NULL`.
This also PG's behaviour:

```
postgres=# \pset null '<null>'
Null display is "<null>".
postgres=# select '{}' != '{}', array_length('{}'::int[], 1) > 0;
 ?column? | ?column? 
----------+----------
 f        | <null>
(1 row)
```

Reply:
>This isn't correct

Right, !=[] works correctly on master.

> array_length requires the same patch.

this still holds true for 
`insert into t values ([null])` followed by `select * from t where array_length(a,1) >0`

Reply:
There are more cases other than `> 0` around `array_length` that fail:

```
cr> select * from t;
+--------------+
| a            |
+--------------+
| [null]       |
| [null, null] |
+--------------+
SELECT 2 rows in set (0.045 sec)
cr> select * from t where array_length(a, 1) = 1;
+---+
| a |
+---+
+---+
SELECT 0 rows in set (0.026 sec)
cr> select * from t where array_length(a, 1) > 1;
+--------------+
| a            |
+--------------+
| [null, null] |
+--------------+
SELECT 1 row in set (0.007 sec)
cr> select * from t where array_length(a, 1) >= 1;
+---+
| a |
+---+
+---+
SELECT 0 rows in set (0.006 sec)
cr> select * from t where array_length(a, 1) < 1;
+---+
| a |
+---+
+---+
SELECT 0 rows in set (0.004 sec)
cr> select * from t where array_length(a, 1) <= 1;
+--------+
| a      |
+--------+
| [null] |
+--------+
SELECT 1 row in set (0.008 sec)

cr> select * from t where array_length(a, 1) >= 0;
+---+
| a |
+---+
+---+
SELECT 0 rows in set (0.009 sec)
```